src/integrationTest/java/org/opensearch/security/SearchOperationTest.java

```diff
@@ -0,0 +1,1348 @@
+			boolQueryBuilder.filter(new MatchQueryBuilder(FIELD_TITLE, TITLE_POISON));
+			searchSourceBuilder.query(boolQueryBuilder);
+			searchRequest.source(searchSourceBuilder);
+
+			assertThatThrownBy(() -> restHighLevelClient.search(searchRequest, DEFAULT), statusException(FORBIDDEN));
+		}
+	}
+
+	@Test
+	public void shouldPerformSearchWithAllIndexAlias_positive() throws IOException {
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(ADMIN_USER)) {
+			SearchRequest searchRequest = queryStringQueryRequest("_all", QUERY_TITLE_MAGNUM_OPUS);
+
+			SearchResponse searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+
+			assertThat(searchResponse, isSuccessfulSearchResponse());
+			assertThat(searchResponse, numberOfTotalHitsIsEqualTo(1));
+			assertThat(searchResponse, searchHitsContainDocumentWithId(0, SONG_INDEX_NAME, ID_S1));
+			assertThat(searchResponse, searchHitContainsFieldWithValue(0, FIELD_TITLE, TITLE_MAGNUM_OPUS));
+		}
+	}
+
+	@Test
+	public void shouldPerformSearchWithAllIndexAlias_negative() throws IOException {
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(LIMITED_READ_USER)) {
+			SearchRequest searchRequest = queryStringQueryRequest("_all", QUERY_TITLE_MAGNUM_OPUS);
+
+			assertThatThrownBy(() -> restHighLevelClient.search(searchRequest, DEFAULT), statusException(FORBIDDEN));
+		}
+	}
+
+	@Test
+	public void shouldScrollOverSearchResults_positive() throws IOException {
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(LIMITED_READ_USER)) {
+			SearchRequest searchRequest = searchRequestWithScroll(SONG_INDEX_NAME, 2);
+			SearchResponse searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+			assertThat(searchResponse, isSuccessfulSearchResponse());
+			assertThat(searchResponse, containNotEmptyScrollingId());
+
+			SearchScrollRequest scrollRequest = getSearchScrollRequest(searchResponse);
+
+			SearchResponse scrollResponse = restHighLevelClient.scroll(scrollRequest, DEFAULT);
+			assertThat(scrollResponse, isSuccessfulSearchResponse());
+			assertThat(scrollResponse, containNotEmptyScrollingId());
+			assertThat(scrollResponse, numberOfTotalHitsIsEqualTo(3));
+			assertThat(scrollResponse, numberOfHitsInPageIsEqualTo(1));
+		}
+	}
+
+	@Test
+	public void shouldScrollOverSearchResults_negative() throws IOException {
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(DOUBLE_READER_USER)) {
+			SearchRequest searchRequest = searchRequestWithScroll(SONG_INDEX_NAME, 2);
+			SearchResponse searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+			assertThat(searchResponse, isSuccessfulSearchResponse());
+			assertThat(searchResponse, containNotEmptyScrollingId());
+
+			SearchScrollRequest scrollRequest = getSearchScrollRequest(searchResponse);
+
+			assertThatThrownBy(() -> restHighLevelClient.scroll(scrollRequest, DEFAULT), statusException(FORBIDDEN));
```

Comment:
The result of request `restHighLevelClient.scroll(scrollRequest, DEFAULT)` is HTTP response with 403 status code because the `DOUBLE_READER_USER` is lacking cluster permission `indices:data/read/scroll`.